### DISCUSSION TITLE
additional-scrape-configs

### DISCUSSION THREAD

**Discussion by hgsat123** [prometheus-operator/prometheus-operator#5265] [Q&A]
Hi,
I deployed prometheus-operator in AWS EKS. i have a requirement to add additional remote-write config which enables prometheus scrapped metrics to AMP (Amazon Managed Prometheus). If i add any additional static scrape config in "additional-scrape-configs.yaml" & create a secret, config-reloader able to update prometheus.env.yaml under /etc/prometheus/config_out to allow prometheus to scrape. If i add a remote-write code block in "additional-scrape-configs.yaml", same not gets updated or config-reloader will not update prometheus.env.yaml.   How to achieve this requirement?  Please clarify

Requirement is: prometheus.env.yaml should have additional remote-write: block code with details of AMP to push metrics there.

**Comment by simonpasquier** [comment]
Please use the `spec.remoteWrite` field for this.

**Comment by hgsat123** [comment]
Hello Simon,

My "additional-scrape-configs.yaml" has below.

remote_write:
- url: https://aps-workspaces.us-east-1.amazonaws.com/workspaces/ws-********/api/v1/remote_write
  remote_timeout: 30s
  write_relabel_configs:
  - action: keep
    regex: (.+)
    source_labels:
    - __name__
  - action: replace
    regex: (.+)
    replacement: aws
    source_labels:
    - __name__
    target_label: ​***_provider.
    .
    .
  - action: labeldrop
    regex: (app_kubernetes_io_.*|helm_.*|beta_kubernetes_io_.*|topology_kubernetes_io_.*|kubernetes_io_.*|failure_domain_beta_kubernetes_io.*)
    tls_config:
      insecure_skip_verify: true
    queue_config:
      batch_send_deadline: 5s
      capacity: 500
      max_backoff: 100ms
      max_samples_per_send: 100
      max_shards: 1000
      min_backoff: 30ms
      min_shards: 1
    metadata_config:
      send_interval: 1m
      max_samples_per_send: 300

We did use "prometheus" earlier & able to use "prometheus-extra-config" to update this additional remote-write into prometheus.yaml (config). I'm trying to achieve that with prometheus-operator.

If i put above block of text into ""additional-scrape-configs.yaml"  & create the secret with name "additional-scrape-configs", config-reloader ignores update. If i use any dummy static scrape config, it works!!

Regards
Satish
________________________________

**Comment by simonpasquier** [comment]
you can't put remote write configurations in the additional scrape config secret (as the name says, it's for scrape configurations only). As I said earlier, you need to use the `remoteWrite` field (see https://prometheus-operator.dev/docs/operator/api/#monitoring.coreos.com/v1.RemoteWriteSpec):

```
apiVersion: monitoring.coreos.com/v1
kind: Prometheus
metadata:
  name: example
spec:
  remoteWrite:
  - url: http://example.com
    ...
```

**Reply by hgsat123** [reply]
oh! ok, Thanks for your clarification. Will try that!

**Reply by hgsat123** [reply]
Hi Simon,
Sorry to bug with one more additional question on this remoteWrite.  After updating  "remoteWrite" field with all details in Prometheus API, do i need to refresh prometheus instances for the changes "remote_write" filed to get updated in prometheus.env.yaml file under config_out? Bcoz, i'm not seeing the config reloader updating the above config file after modifying prometheus API.
Thanks for your time to clarify this.

**Reply by simonpasquier** [reply]
the change should be propagated automatically (the operator generates the Prometheus configuration, the config-reloader only reloads the on-disk configuration when it detects a change). If you don't see the change reflected in the Prometheus configuration, make sure to check the operator's logs.
